### ISSUE TITLE
CSV Field Markers are not collected if they are in another go file

### ISSUE THREAD

**Issue by shivanshs9** [operator-framework/operator-sdk#3744]
## Bug Report

**What did you do?**
In my operator project, CRD status struct is in another go file, called `<CRD>_status.go`. I've annotated some of its fields with operator-sdk CSV markers, like:
```go
type ClusterStatus struct {
	// +operator-sdk:csv:customresourcedefinitions:type=status,displayName="Phase",xDescriptors="urn:alm:descriptor:io.kubernetes.phase"
	Phase ClusterPhase `json:"phase"`
	// [other fields]
}
```
Then I ran `make bundle` to generate the bundle manifests.

**What did you expect to see?**
The generated CSV would have both `specDescriptors` and `statusDescriptors` for this API.

**What did you see instead? Under which circumstances?**
The generated CSV only has the `specDescriptors` of this API and no `statusDescriptors` field.

**Environment**
<details>
<summary>operator-sdk version</summary>

```
operator-sdk version: "v1.0.0", commit: "d7d5e0cd6cf5468bb66e0849f08fda5bf557f4fa", kubernetes version: "v1.18.2", go version: "go1.13.11 linux/amd64", GOOS: "linux", GOARCH: "amd64"
```

</details>

* operator-sdk version: v1.0.0

* go version: go1.13.11

* Kubernetes version information: v1.18.2

* Kubernetes cluster kind: 

* Are you writing your operator in ansible, helm, or go? go

**Possible Solution**

**Additional context**

**Comment by estroz** [comment]
@shivanshs9 is `ClusterStatus` referenced in your API type, like
```go
type Cluster struct {
	v1.TypeMeta   `json:",inline"`
	v1.ObjectMeta `json:"metadata,omitempty"`

	ClusterSpec   `json:"spec,omitempty"`
	ClusterStatus `json:"status,omitempty"`
}
```

**Comment by shivanshs9** [comment]
@estroz Yes, it is indeed referred there. Plus, the CSV works in OpenShift when I manually add the statusDescriptors after generating it.

**Comment by estroz** [comment]
@shivanshs9 does the scenario in #4196 describe the yours as well?

**Comment by shivanshs9** [comment]
> @shivanshs9 does the scenario in #4196 describe the yours as well?


@estroz not exactly, since in my case, only `statusDescriptors` weren't being collected. But I did use a different version, v1.0.0. Haven't tested yet with v1.1.0

**Comment by IBMRob** [comment]
We are also suffering from this problem, We have a number of common structures (in a `common_types.go` file)  which are used in multiple CRDs. None of the `specDescriptors` in the common structures are being generated into the appropriate CRDs. 

This is causing us a major head-ache because we can't share structures across multiple CRDs meaning we need to duplicate lots of code.

**Comment by estroz** [comment]
Reopening this as it has not been resolved by #4445
